src/main/java/com/genesys/workspace/TargetsApi.java

```diff
@@ -43,10 +51,123 @@ public TargetSearchResult search(String searchTerm, TargetsSearchOptions options
                     targets.add(new Target(t));
                 }
             }
-            return new TargetSearchResult(data.getTotalMatches(), targets);
+            return new SearchResult(data.getTotalMatches(), targets);
 
         } catch (ApiException e) {
             throw new WorkspaceApiException("searchTargets failed.", e);
         }
     }
+
+    public void addRecentTarget(TargetInformation target, RecentData recentData) throws WorkspaceApiException {
+        try {
+            TargetsrecentsaddData data = new TargetsrecentsaddData();
+            data.setTarget(target);
+            data.setRecentInformation(recentData);
+            RecentTargetData recentTargetData = new RecentTargetData();
+            recentTargetData.setData(data);
+            ApiSuccessResponse resp = targetsApi.addRecentTarget(recentTargetData);
+            Util.throwIfNotOk(resp);
+        }
+        catch(ApiException ex) {
+            throw new WorkspaceApiException("Cannot add recent target", ex);
+        }
+    }
+
+    public SearchResult<Target> getRecentTargets(int limit) throws WorkspaceApiException {
+        try {
+            TargetsResponse resp = targetsApi.getRecentTargets(new BigDecimal(limit));
+            Util.throwIfNotOk(resp.getStatus());
+            
+            TargetsResponseData data = resp.getData();
+
+            int total = 0;
+            List<Target> list = new ArrayList<>();
+            if(data != null) {
+                total = data.getTotalMatches();
+                if(data.getTargets() != null) {
+                    for (com.genesys._internal.workspace.model.Target t : data.getTargets()) {
+                        Target target = new Target(t);
+                        list.add(target);
+                    }
+                }
+            }
+
+            return new SearchResult(total, list);
+        }
+        catch(ApiException ex) {
+            throw new WorkspaceApiException("Cannot get recent targets", ex);
+        }
+    }
+
+//    public Target getTarget(long id, TargetType type) throws WorkspaceApiException {
+//        try {
+//            ApiSuccessResponse resp = targetsApi.getTarget(new BigDecimal(id), type.getValue());
+//            Util.throwIfNotOk(resp);
+//            
+//        }
+//        catch(ApiException ex) {
+//            throw new WorkspaceApiException("Cannot get target", ex);
+//        }
+//    }
+    
+    public void deletePersonalFavorite(long id, TargetType type) throws WorkspaceApiException {
```

Comment:
I think we could accept just Target here also and then get/save/delete all work in terms of the same object.